### PR TITLE
Implement packaged mode for spring boot jar 

### PR DESCRIPTION
**Pull request by mpeddada1**
Simply copies the jar onto the container. 

Note that this PR also contains some minor naming changes. 

For #2848

### CODE REVIEW

**Comment by mpeddada1** [comment]
This PR is ready for review.

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java [hunk 1]
```diff
@@ -265,6 +282,16 @@ static JarType determineJarType(Path jarPath) throws IOException {
         "java", "-cp", APP_ROOT.toString(), "org.springframework.boot.loader.JarLauncher");
   }
 
+  /**
+   * Computes the entrypoint for a spring boot jar in packaged mode.
+   *
+   * @param jarPath path to jar file
+   * @return list of {@link String} representing entrypoint
+   */
+  static ImmutableList<String> computeEntrypointForPackagedSpringboot(Path jarPath) {
```
Comment (chanseokoh) [inline jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java:291]:
```suggestion
  static ImmutableList<String> computeEntrypointForPackagedSpringBoot(Path jarPath) {
```

File: jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessorTest.java [hunk 2]
```diff
@@ -578,4 +578,20 @@ public void testCreateLayersForExplodedNonLayeredSpringBoot()
             AbsoluteUnixPath.get("/app/BOOT-INF/classes/class1.class"),
             AbsoluteUnixPath.get("/app/BOOT-INF/classes/classDirectory/class2.class"));
   }
+
+  @Test
+  public void testCreateLayersForPackagedSpringboot() throws IOException, URISyntaxException {
```
Comment (chanseokoh) [inline jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessorTest.java:583]:
```suggestion
  public void testCreateLayersForPackagedSpringBoot() throws IOException, URISyntaxException {
```

And probably `IOException` is unnecessary?

File: jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java [hunk 3]
```diff
@@ -212,6 +213,22 @@ static JarType determineJarType(Path jarPath) throws IOException {
...
   }
 
+  /**
+   * Creates layer for the spring-boot jar on the container.
+   *
+   * @param jarPath path to jar file
+   * @return list of {@link FileEntriesLayer}
+   * @throws IOException if I/O error occurs when opening the jar file
+   */
+  static List<FileEntriesLayer> createLayerForPackagedSpringBoot(Path jarPath) throws IOException {
```
Comment (chanseokoh) [inline jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java:223]: Does it throw `IOException`?

File: jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessorTest.java [hunk 4]
```diff
@@ -578,4 +578,20 @@ public void testCreateLayersForExplodedNonLayeredSpringBoot()
...
+    Path springbootJar = Paths.get(Resources.getResource(SPRING_BOOT_JAR).toURI());
+    List<FileEntriesLayer> layers =
+        JarModeProcessor.createLayerForPackagedSpringBoot(springbootJar);
```
Comment (chanseokoh) [inline jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessorTest.java:586]:
```suggestion
        JarModeProcessor.createLayerForPackagedSpringBoot(springBootJar);
```

**Review by chanseokoh — APPROVED** [review]

**Pull request merged**